sklearn/tree/_criterion.pyx

```diff
@@ -199,6 +180,63 @@ cdef class Criterion:
                                  - (self.weighted_n_left /
                                     self.weighted_n_node_samples * impurity_left)))
 
+    cdef void set_sample_pointers(
+        self,
+        SIZE_t start,
+        SIZE_t end
+    ) nogil:
+        """Placeholder for a method which will set sample pointers in the criterion.
```

Comment:
```suggestion
        """Abstract method which will set sample pointers in the criterion.

The dataset array that we compute criteria on is assumed to consist of 'N' ordered samples or rows (i.e. sorted). Since we pass this by reference, we use sample pointers to move the start and end around to consider only a subset of data. 

This function should also update relevant statistics that the class uses to compute the final criterion.
```

^ line spacing prolly needs to be fixed.